### PR TITLE
Migrate `simd-ffi` `run-make` test to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

try-job: x86_64-msvc
try-job: x86_64-mingw
try-job: i686-msvc
try-job: armhf-gnu
try-job: test-various
try-job: aarch64-apple
try-job: x86_64-gnu-llvm-17

### CODE REVIEW

File: tests/run-make/simd-ffi/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+fn main() {
+    let mut targets = Vec::new();
+    // arm-specific targets.
+    if llvm_components_contain("arm") {
+        targets.append(&mut vec![
+            "arm-linux-androideabi".to_owned(),
+            "arm-unknown-linux-gnueabihf".to_owned(),
+            "arm-unknown-linux-gnueabi".to_owned(),
+        ]);
```
Comment (jieyouxu) [inline tests/run-make/simd-ffi/rmake.rs:18]: Nit: this can be `targets.append(&[])` (ref-to-array) right?

File: tests/run-make/simd-ffi/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    let mut x86_archs = Vec::new();
+    if llvm_components_contain("x86") {
+        x86_archs.append(&mut vec!["i686", "x86_64"]);
+    }
```
Comment (jieyouxu) [inline tests/run-make/simd-ffi/rmake.rs:23]: Nit: for this and the follow, I don't think this needs `&mut Vec`, it can just be `&[]` right?

**Review by jieyouxu — APPROVED** [review]
Thanks, LGTM for the most parts, I'll run try jobs to check.

**Comment by jieyouxu** [comment]
r=me after the nits

@bors delegate+

File: tests/run-make/simd-ffi/rmake.rs (hunk 1 above)
Comment (Oneirical) [reply tests/run-make/simd-ffi/rmake.rs:18]: I don't think so:

```
     = note: expected mutable reference `&mut Vec<_>`
                found reference `&[std::string::String; 3]`
```

I've tried various permutations of this, it seems to accept nothing except what I currently have. There might be some clever work-around I'm not aware of, but I will r= the PR for now. Feel free to stop it or invite a revision in a further PR if I misunderstood something.

**Pull request merged**